src/main/java/com/example/processmanagementtool/prospect/ProspectService.java

```diff
@@ -0,0 +1,98 @@
+package com.example.processmanagementtool.prospect;
+
+import com.example.processmanagementtool.domain.prospect.Prospect;
+import com.example.processmanagementtool.domain.prospect.repository.ProspectRepository;
+import com.example.processmanagementtool.domain.template.Template;
+import com.example.processmanagementtool.domain.template.repository.TemplateRepository;
+import com.example.processmanagementtool.domain.user.User;
+import com.example.processmanagementtool.domain.user.repository.UserRepository;
+import com.example.processmanagementtool.dto.ProspectPageDTO;
+import com.example.processmanagementtool.dto.ProspectPropertyDTO;
+import com.example.processmanagementtool.dto.ProspectRequestDTO;
+import com.example.processmanagementtool.dto.SuccessResponseDTO;
+import com.example.processmanagementtool.exception.customexceptions.BadRequest;
+import com.example.processmanagementtool.exception.customexceptions.UserNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple2;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class ProspectService {
+
+    private final ProspectRepository prospectRepository;
+    private final TemplateRepository templateRepository;
+    private final UserRepository userRepository;
+
+    public Mono<SuccessResponseDTO> createNewProspect(Principal principal, Mono<ProspectRequestDTO> request) {
+        return userRepository.findUserByLogin(principal.getName())
+                .flatMap(foundUser -> findTemplateAndValidateData(request, foundUser.getId())
+                        .flatMap(validatedData -> mapValidatedRequestToProspectAndSave(validatedData, foundUser.getId())))
+                .switchIfEmpty(Mono.error(new UserNotFoundException()));
+    }
+
+    public Mono<ProspectPageDTO> findAllProspectsByUser(Principal principal, PageRequest pageRequest) {
+        return userRepository.findUserByLogin(principal.getName())
+                .flatMap(foundUser -> getMapAndZipProspects(foundUser, pageRequest))
+                .map(res -> new PageImpl<>(res.getT1(), pageRequest, res.getT2()))
```

Comment:
nie uzywajmy MAP -> psuje to asynchronicznosc flatMap, tu jest przyklad na osobna prywatna metode tworzenie PageImp nastepnie mapping i zwrotka Mono